src/Symfony/Component/DependencyInjection/EnvProviderInterface.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection;
+
+/**
+ * The EnvProviderInterface is implemented by objects that manage environment-like variables.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+interface EnvProviderInterface
+{
+    /**
+     * Returns the value of the given variable as managed by the current instance.
+     *
+     * @param string   $prefix The namespace of the variable
+     * @param string   $name   The name of the variable within the namespace
+     * @param \Closure $getEnv A closure that allows fetching more env vars
+     *
+     * @return mixed|null The value of the given variable or null when it is not found
```

Comment:
> in case of null" you should return $getEnv

or not - that's implementation specific.

> Mixed includes null

true, but looks more clear to me as is, description included, stating the intend and highlighting the special `null` case... unless more people agree, I wouldn't change here